src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -338,6 +354,12 @@ private DriveInfo ReportCommonDiskUsage(string dirType, string directory)
                 }
                 return true;
             }
+            catch (PathTooLongException)
```

Comment:
Update: I'll put the robocopy stuff back in a new PR but with more logging and a timeout on the RoboCopy process.